lib/core/src/Cardano/Wallet.hs

```diff
@@ -1899,6 +1899,7 @@ mkTxMeta ti' blockHeader wState tx cs expiry =
                 , blockHeight = blockHeader ^. #blockHeight
                 , amount = Quantity $ distance amtInps amtOuts
                 , expiry = Just expiry
+                , deposit = Just . Quantity . Coin $ (cs ^. #deposit)
```

Comment:
This should do the trick... however... it doesn't. It still doesn't end up in the database. I couldn't figure out until now why. I followed the control flow and it should be hitting the database.

Reply:
This should be `Nothing` if `cs ^. #deposit == 0`

Reply:
https://github.com/input-output-hk/cardano-wallet/blob/d4187a713364dca2f00a073338cdc65d37693d36/lib/core/src/Cardano/Wallet/Api/Server.hs#L1495-L1553
This is where the `txMeta` gets updated and the handler indeed returns a deposit. At the same time the handle also transmits this txMeta to the database in https://github.com/input-output-hk/cardano-wallet/blob/d4187a713364dca2f00a073338cdc65d37693d36/lib/core/src/Cardano/Wallet.hs#L1924-L1941

But it never shows up in the database.

Reply:
What do you mean "it never shows up" ?

Reply:
> What do you mean "it never shows up" ?

The database field is `null`.

And I found the culprit: https://github.com/input-output-hk/cardano-wallet/blob/d4187a713364dca2f00a073338cdc65d37693d36/lib/core/src/Cardano/Wallet/Primitive/Model.hs#L383-L392

Reply:
The model needs to be able to reconstruct the deposit's amount from the certificates present in the transaction and the active protocol parameters. 

The main issue here is to know what is the right value of the deposit _at the time the transaction was inserted_. We should be able to land back on our feet by checking the delta between inputs and outputs and the explicit transaction fees. With these, it becomes a simple equation with a single unknown.